corehq/apps/cloudcare/views.py

```diff
@@ -256,7 +257,7 @@ def get(self, request, domain, app_id, **kwargs):
             raise Http404()
 
         role = request.couch_user.get_role(domain)
-        if role and not role.permissions.view_web_app(app):
+        if role and not role.permissions.view_web_app((app['copy_of'] or app['_id'])):
```

Comment:
Here you can use `app.master_id` since the app has been [wrapped by jsonobject](https://github.com/dimagi/commcare-hq/blob/e3c6690b97b76ac22561dbd88a96babfb27e91cc/corehq/apps/app_manager/dbaccessors.py#L173)

```suggestion
        if role and not role.permissions.view_web_app(app.master_id):
```